### DISCUSSION TITLE
json and escaping utf-8 characters

### DISCUSSION THREAD

**Discussion by PokerMakyo** [micropython/micropython#16497] [Libraries & Drivers]
I see that micropython json library handles utf-8 characters differently than python3 does. I'm trying to send a request with some json data containing non-ascii characters:
```
requests.get('http://some-url/', json=u'\u0142azienka')
```

The request generated by python3 contains:
```
00d0   20 31 35 0d 0a 43 6f 6e 74 65 6e 74 2d 54 79 70    15..Content-Typ
00e0   65 3a 20 61 70 70 6c 69 63 61 74 69 6f 6e 2f 6a   e: application/j
00f0   73 6f 6e 0d 0a 0d 0a 22 5c 75 30 31 34 32 61 7a   son...."\u0142az
0100   69 65 6e 6b 61 22                                 ienka"
```

while micropython generates:
```
0090   43 6f 6e 74 65 6e 74 2d 54 79 70 65 3a 20 61 70   Content-Type: ap
00a0   70 6c 69 63 61 74 69 6f 6e 2f 6a 73 6f 6e 0d 0a   plication/json..
00b0   0d 0a 22 c5 82 61 7a 69 65 6e 6b 61 22            .."..azienka"
```
Is there a way to force micropython to escape utf-8 characters like python3 does? Do I need to write my own escaping function?
Looks like this is related to: https://github.com/micropython/micropython/issues/4397

**Comment by karfas** [comment]
While I have no idea what's wrong with the micropython request, your python3 dump shows the complete escape sequence including the backslash.
This is for sure also not an UTF-8 character.

**Reply by mendenm** [reply]
Interestingly, it appears that micropython did it right, here...

```
Python 3.11.5 (main, Sep 11 2023, 08:19:27) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> "\u0142"
'ł'
>>> bytearray("\u0142",'utf8')
bytearray(b'\xc5\x82')
>>> 
```

That is, the utf8 representation of unicode point \u0142 is exactly as the micropython output showed in the hexdump.

(Added note: have you tried "Content-type: application/json; charset=utf-8" as the header, to make sure the receiver is decoding correctly?
see, e.g. https://stackoverflow.com/questions/9254891/what-does-content-type-application-json-charset-utf-8-really-mean
)

(Added added note:
Both of the originally shown outputs actually appear to be valid json representation of the original string, see, e.g.
https://stackoverflow.com/questions/19176024/how-to-escape-special-characters-in-building-a-json-string
)
